app/app_client.go

```diff
@@ -0,0 +1,1241 @@
+	resp, err := c.client.GetUserIDByEmail(ctx, &pb.GetUserIDByEmailRequest{
+		Email: email,
+	})
+	if err != nil {
+		return "", err
+	}
+	return resp.UserId, nil
+}
+
+// CreateOrganization creates a new organization.
+func (c *AppClient) CreateOrganization(ctx context.Context, name string) (*Organization, error) {
+	resp, err := c.client.CreateOrganization(ctx, &pb.CreateOrganizationRequest{
+		Name: name,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return organizationFromProto(resp.Organization), nil
+}
+
+// ListOrganizations lists all the organizations.
+func (c *AppClient) ListOrganizations(ctx context.Context) ([]*Organization, error) {
+	resp, err := c.client.ListOrganizations(ctx, &pb.ListOrganizationsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	var organizations []*Organization
+	for _, org := range resp.Organizations {
+		organizations = append(organizations, organizationFromProto(org))
+	}
+	return organizations, nil
+}
+
+// GetOrganizationsWithAccessToLocation gets all the organizations that have access to a location.
+func (c *AppClient) GetOrganizationsWithAccessToLocation(ctx context.Context, locationID string) ([]*OrganizationIdentity, error) {
+	resp, err := c.client.GetOrganizationsWithAccessToLocation(ctx, &pb.GetOrganizationsWithAccessToLocationRequest{
+		LocationId: locationID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var organizations []*OrganizationIdentity
+	for _, org := range resp.OrganizationIdentities {
+		organizations = append(organizations, organizationIdentityFromProto(org))
+	}
+	return organizations, nil
+}
+
+// ListOrganizationsByUser lists all the organizations that a user belongs to.
+func (c *AppClient) ListOrganizationsByUser(ctx context.Context, userID string) ([]*OrgDetails, error) {
+	resp, err := c.client.ListOrganizationsByUser(ctx, &pb.ListOrganizationsByUserRequest{
+		UserId: userID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var organizations []*OrgDetails
+	for _, org := range resp.Orgs {
+		organizations = append(organizations, orgDetailsFromProto(org))
+	}
+	return organizations, nil
+}
+
+// GetOrganization gets an organization.
+func (c *AppClient) GetOrganization(ctx context.Context, orgID string) (*Organization, error) {
+	resp, err := c.client.GetOrganization(ctx, &pb.GetOrganizationRequest{
+		OrganizationId: orgID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return organizationFromProto(resp.Organization), nil
+}
+
+// GetOrganizationNamespaceAvailability checks for namespace availability throughout all organizations.
+func (c *AppClient) GetOrganizationNamespaceAvailability(ctx context.Context, namespace string) (bool, error) {
+	resp, err := c.client.GetOrganizationNamespaceAvailability(ctx, &pb.GetOrganizationNamespaceAvailabilityRequest{
+		PublicNamespace: namespace,
+	})
+	if err != nil {
+		return false, err
+	}
+	return resp.Available, nil
+}
+
+// UpdateOrganizationOptions contains optional parameters for UpdateOrganization.
```

Comment:
Agree! I'm surprised the linter isn't yelling at you, actually, as it will usually require an "export comment" on every exported field of an exported struct.

Reply:
Yeah, the linter requires you to have a comment for each struct, which is why that comment is there on all of them. But yes, will definitely look through one more time to see confusing types.